geode-core/src/main/java/org/apache/geode/internal/cache/entries/AbstractRegionEntry.java

```diff
@@ -931,6 +910,30 @@ public boolean destroy(InternalRegion region, EntryEventImpl event, boolean inTo
     }
   }
 
+  protected void updateIndexOnDestroyOperation(InternalRegion region) {
+    if (!isTombstone() && !region.isProxy() && !isInvalid()) {
```

Comment:
Probably doesn't matter much but do we want to still have the region proxy check first?